updatehub/src/states/poll.rs

```diff
@@ -28,110 +28,93 @@ impl StateChangeImpl for State<Poll> {
         shared_state: &mut SharedState,
     ) -> Result<(StateMachine, actor::StepTransition)> {
         crate::logger::start_memory_logging();
-        let current_time: DateTime<Utc> = Utc::now();
 
         if shared_state.runtime_settings.is_polling_forced() {
             debug!("Moving to Probe state as soon as possible.");
             return Ok((StateMachine::Probe(self.into()), actor::StepTransition::Immediate));
         }
 
-        let last_poll = shared_state.runtime_settings.last_polling();
+        let interval = Utc::now()
+            .signed_duration_since(shared_state.runtime_settings.last_polling())
+            - shared_state.settings.polling.interval;
 
-        if last_poll > current_time {
-            info!("Forcing to Probe state as last polling seems to happened in future.");
+        if interval.num_seconds() < 0 {
```

Comment:
I believe the logic is inverted. We should trigger it when we have spent more time since the last polling, so it'll be bigger than 0.